python/graphscope/nx/algorithms/builtin.py

```diff
@@ -1045,6 +1048,97 @@ def edge_boundary(G, nbunch1, nbunch2=None):
     return ctx.to_numpy("r", axis=0).tolist()
 
 
+@project_to_simple
+def attribute_assortativity_coefficient(G, attribute):
+    """Compute assortativity for node attributes.
+
+    Assortativity measures the similarity of connections
+    in the graph with respect to the given attribute.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+
+    attribute : string
+        Node attribute key
+
+    Returns
+    -------
+    r: float
+       Assortativity of graph for given attribute
+
+    Examples
+    --------
+    >>> G = nx.Graph()
+    >>> G.add_nodes_from([0, 1], color="red")
+    >>> G.add_nodes_from([2, 3], color="blue")
+    >>> G.add_edges_from([(0, 1), (2, 3)])
+    >>> print(nx.builtin.attribute_assortativity_coefficient(G, "color"))
+    1.0
+
+    Notes
+    -----
+    This computes Eq. (2) in Ref. [1]_ , (trace(M)-sum(M^2))/(1-sum(M^2)),
+    where M is the joint probability distribution (mixing matrix)
+    of the specified attribute.
+
+    References
+    ----------
+    .. [1] M. E. J. Newman, Mixing patterns in networks,
+       Physical Review E, 67 026126, 2003
+    """
+
+    ctx = AppAssets(algo="attribute_assortativity_coefficient", context="tensor")(
```

Comment:
like [degree_assortativity_coefficient](https://github.com/alibaba/GraphScope/blob/13b6b4f87874fb0c57a5d3d2dbfe703dd6027b9c/python/graphscope/nx/algorithms/builtin.py#L950), here we shoule call `graphscope.attribute_assortativity_coefficient`